tests/ui/filter_map_next.rs

```diff
@@ -0,0 +1,13 @@
+#![warn(clippy::all, clippy::pedantic)]
+
+fn main() {
+    let a = ["1", "lol", "3", "NaN", "5"];
+
+    let element: Option<i32> = a.iter().filter_map(|s| s.parse().ok()).next();
+    assert_eq!(element, Some(1));
+
+    let _: Option<u32> = vec![1, 2, 3, 4, 5, 6]
+        .into_iter()
+        .filter_map(|x| if x == 2 { Some(x * 2) } else { None })
```

Comment:
To test this, the argument needs to be multiline:
```suggestion
        .filter_map(|x| if x == 2 { 
            Some(x * 2) 
        } else { 
            None 
        })
```

Reply:
Ops! I did that actually, but lost it after rustfmt. I'm skipping it now. 

```
   = note: replace `filter_map(|x| {
                       if x == 2 {
                           Some(x * 2)
                       } else {
                           None
                       }
                   }).next()` with `find_map(|x| {
                       if x == 2 {
                           Some(x * 2)
                       } else {
                           None
                       }
                   })`
```
Is making a 'big' suggestion like this ok? Depending on the size of the actual code, maybe it's better to just do something like `"replace filter_map(p).next() with find_map(p)" ?

Other fns on this file are doing something close to that. Like `lint_filter_next`